File: packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.Input/src/InputTypes/OperationResponse.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;

namespace Microsoft.TypeSpec.Generator.Input
{
    /// <summary>
    /// Represents an operation response.
    /// </summary>
    public sealed class OperationResponse
    {
        public OperationResponse(IReadOnlyList<int> statusCodes, InputType? bodyType, IReadOnlyList<OperationResponseHeader> headers, bool isErrorResponse, IReadOnlyList<string> contentTypes)
        {
            StatusCodes = statusCodes;
            BodyType = bodyType;
            Headers = headers;
            IsErrorResponse = isErrorResponse;
            ContentTypes = contentTypes;
        }

        public OperationResponse() : this(Array.Empty<int>(), null,  Array.Empty<OperationResponseHeader>(), false, Array.Empty<string>()) { }

        public IReadOnlyList<int> StatusCodes { get; }
        public InputType? BodyType { get; }
        public IReadOnlyList<OperationResponseHeader> Headers { get; }
        public bool IsErrorResponse { get; }
        public IReadOnlyList<string> ContentTypes { get; }
    }
}

File: packages/http-client-csharp/emitter/src/type/operation-response.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import type { SerializationOptions } from "@azure-tools/typespec-client-generator-core";
import type { HttpResponseHeader } from "./http-response-header.js";
import type { InputType } from "./input-type.js";

export interface OperationResponse {
  statusCodes: number[];
  bodyType?: InputType;
  headers: HttpResponseHeader[];
  contentTypes?: string[];
  isErrorResponse: boolean;
  serializationOptions: SerializationOptions;
}
